ffi/src/lib.rs

```diff
@@ -131,6 +131,23 @@ impl KernelStringSlice {
     }
 }
 
+/// FFI-safe implementation for Rust's Option<T>
+#[derive(PartialEq, Debug)]
+#[repr(C)]
+pub enum OptionalValue<T> {
+    Ok(T),
```

Comment:
considering `Ok` is typically associated with `Result`s perhaps we should just match the [`Option`'s](https://doc.rust-lang.org/std/option/enum.Option.html) `Some`?
```suggestion
    Some(T),
```